sdks/python/apache_beam/runners/worker/statesampler.py

```diff
@@ -46,13 +64,24 @@ def __init__(self, prefix, counter_factory,
     self._prefix = prefix
     self._counter_factory = counter_factory
     self._states_by_name = {}
+    self._registered = False
     self.sampling_period_ms = sampling_period_ms
+
+    # This keeps track of the MetricsContainer instances of the work item
+    # tracked by this sampler.
+    self.metrics_container_registry = {}
+    self.register()
     super(StateSampler, self).__init__(sampling_period_ms)
 
   def stop_if_still_running(self):
     if self.started and not self.finished:
       self.stop()
 
+  def register(self):
```

Comment:
Maybe rename this register_for_thread? I think it'd be clearer for the caller to instead call something like

    statesampler.set_current_sampler(sampler_instance)

than

    sampler_instance.register[_for_thread]()